sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -160,6 +160,63 @@ static rom_error_t boot_data_entry_read(uint32_t page_base, size_t index,
   return kErrorOk;
 }
 
+/**
+ * Populates the boot data entry at the given page and index.
+ *
+ * @param page_base Page base address in bytes.
+ * @param index Index of the entry to write in the given page.
+ * @param boot_data Entry to write.
+ * @return The result of the operation.
+ */
+static rom_error_t boot_data_entry_write(uint32_t page_base, size_t index,
+                                         const boot_data_t *boot_data) {
+  const uint32_t addr = boot_data_entry_address_get(page_base, index);
+  boot_data_buffer_t buf;
+  memcpy(&buf, boot_data, sizeof(boot_data_t));
+  // TODO(#8777): Update error handling after switching to silicon_creator
+  // driver.
+  if (flash_write(addr, kInfoPartition, buf.data, kBootDataNumWords) != 0) {
+    return kErrorBootDataFlash;
+  }
+  if (flash_read(addr, kInfoPartition, kBootDataNumWords, buf.data) != 0) {
+    return kErrorBootDataFlash;
```

Comment:
Currently, yes. I will keep this in mind when migrating to the silicon_creator flash_ctrl driver. I guess it's better to make the driver return the appropriate return code instead of translating it here.